test/integration/targets/nxos_user/tests/common/sanity.yaml

```diff
@@ -0,0 +1,116 @@
+---
+- debug: msg="START connection={{ ansible_connection }} nxos_user parameter test"
+- debug: msg="Using provider={{ connection.transport }}"
+  when: ansible_connection == "local"
+
+- set_fact: idem="true"
+- set_fact: idem="false"
+  when: ((platform is search('N7K')) and (imagetag and (imagetag is version_compare('D1', 'eq'))))
+
+- block:
+  - name: Create user
+    nxos_user: &configure
+      name: netend
+      configured_password: Hello!23$
+      update_password: on_create
+      roles: network-operator
+      state: present
+      provider: "{{ connection }}"
+    register: result
+
+  - assert: &true
+      that:
+        - 'result.changed == true'
+
+  - block:
+    - name: conf idempotency
+      nxos_user: *configure
+      register: result
+
+    - assert: &false
+        that:
+          - 'result.changed == false'
+    when: idem
+
+  - name: Remove user
+    nxos_user: &remove
+      name: netend
+      state: absent
+      provider: "{{ connection }}"
+    register: result
+
+  - assert: *true
+
+  - name: remove idempotency
+    nxos_user: *remove
+    register: result
+
+  - assert: *false
+
+  - debug: msg="skipping sshkey test as the key needs to be created on the server first"
+
+#  - name: create a new user
+#    nxos_user: &conf1
+#      name: ansible
+#      sshkey: "{{ lookup('file', '~/.ssh/id_rsa.pub') }}"
+#      state: present
+#      provider: "{{ connection }}"
+#    register: result
+#
+#  - assert: *true
+
+#  - block:
+#    - name: user idempotency
+#      nxos_user: *conf1
+#      register: result
+#
+#    - assert: *false
+#
+#    when: idem
+
+  - name: Collection of users
+    nxos_user: &coll
+      users:
+        - name: test1
+        - name: test2
+      configured_password: Hello!23$
+      update_password: on_create
+      state: present
+      roles: 
+        - network-admin
+        - network-operator
+      provider: "{{ connection }}"
+    register: result
+
+  - assert: *true
+
+  - block:
+    - name: users idempotency
+      nxos_user: *coll
+      register: result
+
+    - assert: *false
+
+    when: idem
+
+  - name: tearDown
+    nxos_user: &tear
+      purge: yes
```

Comment:
Instead of using `purge`, please use `aggregate` with state:absent to delete the users created as `purge` tries to delete account with admin role which cause idempotence issue.

Reply:
Not really. purge actually skips admin user. It is in the code as well as documented properly.
See this for dpcumentation:
https://github.com/ansible/ansible/blob/cdd21e2170b9a47d84624c8c2f792a66f2b0fe57/lib/ansible/modules/network/nxos/nxos_user.py#L83
 
Also here is the code:
https://github.com/ansible/ansible/blob/cdd21e2170b9a47d84624c8c2f792a66f2b0fe57/lib/ansible/modules/network/nxos/nxos_user.py#L362

I added this to test 'purge' functionality only. If we remove this, there is no test for 'purge'. Also in basic.yaml, the users are being deleted using aggregate so that functionality is being tested as well.

Reply:
yeah, the test looks good for CI. so in our lab, we have `ansible` as admin user. this will cause idempotence failure when we test on lab, you can do something like:
```
diff --git a/test/integration/targets/nxos_user/tests/common/sanity.yaml b/test/integration/targets/nxos_user/tests/common/sanity.yaml
index 274548ab61..a0ae4ddb47 100644
--- a/test/integration/targets/nxos_user/tests/common/sanity.yaml
+++ b/test/integration/targets/nxos_user/tests/common/sanity.yaml
@@ -95,6 +95,8 @@
 
   - name: tearDown
     nxos_user: &tear
+     name: ansible
       purge: yes
       provider: "{{ connection }}"
     register: result
```
this will leave `ansible` user if it exists and remove rest of the users as code doesn't handle it, since we are not doing any test against `ansible` user, leaving it shouldn't cause any problem.

Reply:
Done.

Reply:
Thanks.